inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/core/operation/InlongStreamProcessOperation.java

```diff
@@ -0,0 +1,291 @@
+                    processForm);
+            ProcessStatus processStatus = workflowResult.getProcessInfo().getStatus();
+            return processStatus == ProcessStatus.COMPLETED;
+        } else {
+            executorService.execute(
+                    () -> workflowService.start(processName, operator, processForm));
+            return true;
+        }
+    }
+
+    /**
+     * Suspend stream in synchronous/asynchronous way.
+     *
+     * @param groupId
+     * @param streamId
+     * @param operator
+     * @param sync
+     * @return
+     */
+    public boolean suspendProcess(String groupId, String streamId, String operator, boolean sync) {
+        log.info("SuspendProcess for groupId={}, streamId={}", groupId, streamId);
+        InlongGroupInfo groupInfo = groupService.get(groupId);
+        if (groupInfo == null) {
+            throw new BusinessException(ErrorCodeEnum.GROUP_NOT_FOUND);
+        }
+        GroupStatus groupStatus = GroupStatus.forCode(groupInfo.getStatus());
+        if (groupStatus != GroupStatus.CONFIG_SUCCESSFUL
+                && groupStatus != GroupStatus.RESTARTED
+                && groupStatus != GroupStatus.SUSPENDED) {
+            throw new BusinessException(
+                    String.format("GroupId=%s, status=%s not correct for stream suspend", groupId, groupStatus));
+        }
+        InlongStreamInfo streamInfo = streamService.get(groupId, streamId);
+        if (streamInfo == null) {
+            throw new BusinessException(ErrorCodeEnum.STREAM_NOT_FOUND);
+        }
+        StreamStatus status = StreamStatus.forCode(streamInfo.getStatus());
+        if (status == StreamStatus.SUSPENDED || status == StreamStatus.SUSPENDING) {
+            log.warn("GroupId={}, StreamId={} is already in {}", groupId, streamId, status);
+            return true;
+        }
+        if (status != StreamStatus.CONFIG_SUCCESSFUL && status != StreamStatus.RESTARTED) {
+            throw new BusinessException(
+                    String.format("GroupId=%s, StreamId=%s, status=%s not correct for stream suspend", groupId,
+                            streamId,
+                            status));
+        }
+        StreamResourceProcessForm processForm = genStreamProcessForm(groupInfo, streamInfo, GroupOperateType.SUSPEND);
+        ProcessName processName = ProcessName.SUSPEND_STREAM_RESOURCE;
+        if (sync) {
+            WorkflowResult workflowResult = workflowService.start(processName, operator,
+                    processForm);
+            ProcessStatus processStatus = workflowResult.getProcessInfo().getStatus();
+            return processStatus == ProcessStatus.COMPLETED;
+        } else {
+            executorService.execute(
+                    () -> workflowService.start(processName, operator, processForm));
+            return true;
+        }
+    }
+
+    /**
+     * Restart stream in synchronous/asynchronous way.
+     *
+     * @param groupId
```

Comment:
If the params and return tag were empty, just remove them.

Reply:
done